nemo/backends/pytorch/tutorials/toys.py

```diff
@@ -154,22 +154,19 @@ def __init__(self, batch_size, f_name="sin", n=1000, x_lo=-4, x_hi=4):
 
         self._n = n
         self._batch_size = batch_size
-        self._device = t.device("cuda" if self.placement == DeviceType.GPU else "cpu")
 
-        x_data = t.tensor(np.random.uniform(low=x_lo, high=x_hi, size=self._n)).unsqueeze(-1).to(self._device)
+        x_data = t.tensor(np.random.uniform(low=x_lo, high=x_hi, size=self._n)).unsqueeze(-1)
         y_data = func(x_data)
-
-        self._data_iterator = t_utils.DataLoader(
-            t_utils.TensorDataset(x_data.float(), y_data.float()), batch_size=self._batch_size,
-        )
+        self._dataset = t_utils.TensorDataset(x_data.float(), y_data.float())
+        self._data_iterator = t_utils.DataLoader(self._dataset, batch_size=self._batch_size,)
 
     @property
     def data_iterator(self):
         return self._data_iterator
 
     @property
     def dataset(self):
-        return None
+        return self._dataset
```

Comment:
why not both? The problem is some other tests/scripts are using the data_iterator functionality of this class.

Reply:
we've been saying that DataLayerNM should implement either one but not both https://nvidia.github.io/NeMo/tutorials/custommodules.html#example 
The reason for this is when both are implemented, how does a training API should behave - e.g. which one should it use? Maybe a reasonable thing would be to try using data_iterator first (if implemented).
It's probably fine to keep both, but could you please create a JIRA or issue to make sure we clarify the behavior and update the docs accordingly in case when both are implemented

Reply:
i agree if iterator is implemented, we can see it as 'overwriting' default iterator behavior.  I will create jira issue